lib/benchmarks/gallery_automator.dart

```diff
@@ -0,0 +1,201 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:gallery/data/demos.dart';
+import 'package:gallery/l10n/gallery_localizations_en.dart';
+import 'package:gallery/main.dart';
+
+import 'scroll.dart';
+
+const Duration _initialWaitingDuration = Duration(milliseconds: 1500);
+
+const List<String> _demosWithAnimation = <String>[
+  'Progress indicators@material',
+  'Activity indicator@cupertino',
+  'Colors@other',
+];
+
+const Duration _defaultWaitingDuration = Duration(seconds: 3);
+
+enum DemoType {
+  study,
+  animatedWidget,
+  unanimatedWidget,
+}
+
+DemoType typeOfDemo(String demo) {
+  if (demo.contains('@study')) {
+    return DemoType.study;
+  } else if (_demosWithAnimation.contains(demo)) {
+    return DemoType.animatedWidget;
+  } else {
+    return DemoType.unanimatedWidget;
+  }
+}
+
+/// A class that automates the gallery.
+class GalleryAutomator {
+  GalleryAutomator({
+    @required this.benchmarkName,
+    @required this.reporter,
+    @required this.reportError,
+    @required this.runCriterion,
+    this.testScrollsOnly = false,
+  });
+
+  final String benchmarkName;
+  final void Function(String) reporter;
+  final Future<void> Function(dynamic error, StackTrace stackTrace) reportError;
+  final bool Function(String) runCriterion;
+  final bool testScrollsOnly;
+
+  bool finished = false;
```

Comment:
Where is this consumed?

Reply:
[flutter/flutter — dev/benchmarks/macrobenchmarks/lib/src/web/gallery/gallery_recorder.dart:48
](https://github.com/flutter/flutter/pull/57576/files#diff-25d548fb6463b4b394e412dc939a415fR48)

```
  /// Whether we should continue recording.
  @override
  bool shouldContinue() => !_finished || profile.shouldContinue();

  GalleryAutomator _galleryAutomator;
  bool get _finished => _galleryAutomator?.finished ?? false;
```